tarsier-service/next/src/app/dash/page.tsx

```diff
@@ -0,0 +1,64 @@
+"use client"
+import { useRouter, useSearchParams } from "next/navigation";
+import { withAuthInfo, type WithAuthInfoProps } from "@propelauth/react"
+import { Button } from "@/components/ui/button";
+import LogsTable from "@/components/apiLogsTable"
+import CheckoutForm from "@/components/checkoutForm";
+import { useToast } from "@/components/ui/use-toast"
+import { useEffect, useState } from "react";
+import { BACKEND_BASE } from "@/constants/api";
+
+const Dashboard = withAuthInfo((props: WithAuthInfoProps) => {
+  const requestLimit: number = (props.user as any)?.metadata?.request_limit || 100
+  const router = useRouter()
+  const searchParams = useSearchParams()
+  const { toast } = useToast()
+  const [jobsThisMonth, setJobsThisMonth] = useState<'?' | number>('?')
+
+
+  if (searchParams.get("session_id")) {
+    toast({ "title": "Payment Successful", "description": "API monthly limit set to 1000", "className": "bg-black text-white" })
+    router.replace("/dash");
+  }
+
+  useEffect(() => {
+    const headers = { 'Authorization': `Bearer ${props.accessToken}`}
+    fetch(`${BACKEND_BASE}/job-count-cur-month`, { headers })
+      .then(r => r.json())
+      .then(count => {
+        setJobsThisMonth(count)
+      })
+  }, [])
```

Comment:
ideally down the road we'd use react-query for this

Reply:
https://tanstack.com/query/latest/docs/framework/react/overview